keras/regularizers.py

```diff
@@ -16,6 +17,45 @@ def get_config(self):
         return {'name': self.__class__.__name__}
 
 
+class EigenvalueRegularizer(Regularizer):
+    """This class implements the Eigenvalue Decay regularizer.
+    
+    Args:
+        The constant that controls the regularization on the current layer
+        ( see Section 3 of https://arxiv.org/abs/1604.06985 )
+
+    Returns:
+        The regularized loss (for the training data) and
+        the original loss (for the validation data).
+        
+    """
+    def __init__(self, k):
+        self.k = k
+        self.uses_learning_phase = True
+
+    def set_param(self, p):
+        self.p = p
+
+    def __call__(self, loss):
+        power = 9  # number of iterations of the power method
+        W = self.p
+        WW = K.dot(K.transpose(W), W)
+        dim1, dim2 = K.eval(K.shape(WW))  # number of neurons in the layer
```

Comment:
It is done. The extension of this regularizer to convolutional layers (and the respective analysis) may be an interesting study, I will try to build the framework for this analysis, maybe also based on the classification margin.